src/renderer/containers/App/handleUploadJobUpdates.ts

```diff
@@ -0,0 +1,115 @@
+import { Dispatch } from "react";
+
+import { FSSUpload, UploadStatus } from "../../services/file-storage-service";
+import {
+  JSSJob,
+  Service,
+  UploadJob,
+} from "../../services/job-status-service/types";
+import {
+  receiveFSSJobCompletionUpdate,
+  receiveJobUpdate,
+  updateUploadProgressInfo,
+} from "../../state/job/actions";
+import { Step } from "../Table/CustomCells/StatusCell/Step";
+
+/**
+ * Handles job updates for a standard file upload.
+ * Reports progress on the pre-upload MD5 step, the upload step, and then the post-upload MD5 step.
+ */
+function handleFSSJobUpdate(job: FSSUpload, dispatch: Dispatch<any>) {
+  const totalBytes = job.serviceFields.fileSize || 0; // 0 is a safe default, but in practice filesize is initialized immediately after job creation.
+  if (
+    job.serviceFields?.preUploadMd5 &&
+    job.serviceFields.preUploadMd5 !== job.serviceFields.fileSize
+  ) {
+    dispatch(
+      updateUploadProgressInfo(job.jobId, {
+        bytesUploaded: job.serviceFields?.preUploadMd5,
+        totalBytes,
+        step: Step.ONE,
+      })
+    );
+  } else if (
+    job.serviceFields?.currentFileSize &&
+    job.serviceFields.currentFileSize !== job.serviceFields?.fileSize
+  ) {
+    dispatch(
+      updateUploadProgressInfo(job.jobId, {
+        bytesUploaded: job.serviceFields?.currentFileSize,
+        totalBytes,
+        step: Step.TWO,
+      })
+    );
+  } else if (
+    job.serviceFields?.postUploadMd5 &&
+    job.serviceFields.postUploadMd5 !== job.serviceFields?.fileSize
+  ) {
+    dispatch(
+      updateUploadProgressInfo(job.jobId, {
+        bytesUploaded: job.serviceFields?.postUploadMd5,
+        totalBytes,
+        step: Step.THREE,
+      })
+    );
+  }
+}
```

Comment:
`handleFSSJobUpdate()` is largely just copy-pasted from its original form in `App/index.tsx`, but I'm open to comments / suggestions related to it.